uPortal-layout/uPortal-layout-core/src/main/java/org/apereo/portal/layout/LayoutPortletPersonalizer.java

```diff
@@ -0,0 +1,82 @@
+package org.apereo.portal.layout;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.portal.security.IPerson;
+
+/**
+ * Class that helps personalize portlet definitions by replacing tokens with user attributes.
```

Comment:
Interesting, what are some use cases for customizing layout based on user attributes?

Reply:
It's most often used in 2 cases. First, for setting an ID parameter in URLs for some non-sensitive services. Second, for personalization of text.

Reply:
makes sense, thanks for the context!